framework/infrastructure/traffic_director.py

```diff
@@ -612,30 +631,56 @@ def create_target_proxy(self):
             target_proxy_type,
             self.url_map.name,
         )
-        self.target_proxy = create_proxy_fn(name, self.url_map)
 
-    def delete_target_grpc_proxy(self, force=False):
+        if ipv6:
+            self.target_proxy_v6 = create_proxy_fn(name, self.url_map)
+        else:
+            self.target_proxy = create_proxy_fn(name, self.url_map)
+
+    def delete_target_grpc_proxy(self, force=False, ipv6=False):
         if force:
-            name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            if ipv6:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME_IPV6)
+            else:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME)
+        elif ipv6:
+            if self.target_proxy_v6:
```

Comment:
Will there be a case it's `ipv6` but not `self.target_proxy_v6`? What will be the name?

Reply:
Refactored ipv6 creations and deletions to separate methods